samples/boards/nrf52/power_mgr/Kconfig

```diff
@@ -3,8 +3,8 @@
 config NRF5_POWER_MGMT_EXAMPLE
 	bool
 	default y
+	select HAS_SYS_POWER_STATE_SLEEP_1
 	select HAS_SYS_POWER_STATE_SLEEP_2
```

Comment:
As noted in the PR description the NRF52 doesn't support 'Sleep' states, only 'Deep Sleep 1' state. We shouldn't enable states that are not supported. This is going to be confusing for anyone trying to understand the design of Zephyr power management subsystem. The `HAS_SYS_POWER_STATE_` options should be set by the respective SoC not the sample app. Sample applications are there to provide guidance, showcase usage. This example is incorrect.

When run with `CONFIG_SOC_LOG_LEVEL_DBG=y` the app generates the output
```
3 power states are supported
[  0.036]: Sys states for 03: +S1 +S2
[  0.039]: Busy-wait 1 s
[  1.036]: Sleep 1499 ms
[  2.537]: Sleep 2500 ms
[  2.540]:  --> Entering to SYS_POWER_STATE_SLEEP_1 state.
[  5.040]:  --> Exited from SYS_POWER_STATE_SLEEP_1 state.
[  5.045]: Sleep 3500 ms
[00:00:05.045,440] <dbg> soc._sys_pm_power_state_exit_post_ops: Unsupported power state 0
[00:00:05.047,760] <dbg> soc.sys_set_power_state: Unsupported power state 0
[  6.061]:  --> Entering to SYS_POWER_STATE_SLEEP_1 state.
[  8.547]:  --> Exited from SYS_POWER_STATE_SLEEP_1 state.
[  8.552]: Sys states for 02: -S1 +S2
[  8.556]: Busy-wait 1 s
[  9.552]: Sleep 1499 ms
[00:00:08.552,917] <dbg> soc._sys_pm_power_state_exit_post_ops: Unsupported power state 0
[00:00:09.563,507] <dbg> soc.sys_set_power_state: Unsupported power state 0
[ 11.054]: Sleep 2500 ms
[ 13.556]: Sleep 3500 ms
[ 13.558]:  --> Entering to SYS_POWER_STATE_SLEEP_2 state.
[ 17.058]:  --> Exited from SYS_POWER_STATE_SLEEP_2 state.
[ 17.063]: Sys states for 01: +S1 -S2
[ 17.067]: Busy-wait 1 s
[ 18.063]: Sleep 1499 ms
[00:00:17.063,781] <dbg> soc._sys_pm_power_state_exit_post_ops: Unsupported power state 1
[00:00:18.074,157] <dbg> soc.sys_set_power_state: Unsupported power state 1
```